spark/src/main/scala/geotrellis/spark/io/s3/S3LayerDeleter.scala

```diff
@@ -0,0 +1,54 @@
+package geotrellis.spark.io.s3
+
+import geotrellis.spark.{Boundable, KeyBounds, LayerId}
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.io._
+import org.apache.spark.SparkContext
+import spray.json.JsonFormat
+import geotrellis.spark.io.json._
+import spray.json.DefaultJsonProtocol._
+import scala.reflect.ClassTag
+
+class S3LayerDeleter[K: Boundable: JsonFormat: ClassTag]
+  (attributeStore: AttributeStore[JsonFormat])(implicit sc: SparkContext) extends LayerDeleter[LayerId] {
+
+  def getS3Client: () => S3Client = () => S3Client.default
+
+  def delete(id: LayerId): Unit = {
+    if (!attributeStore.layerExists(id)) throw new LayerNotFoundError(id)
+    val (header, _, keyBounds, keyIndex, _) = try {
+      attributeStore.readLayerAttributes[S3LayerHeader, Unit, KeyBounds[K], KeyIndex[K], Unit](id)
+    } catch {
+      case e: AttributeNotFoundError => throw new LayerDeleteError(id).initCause(e)
+    }
+
+    val bucket = header.bucket
+    val prefix = header.key
+    val numPartitions = 1
```

Comment:
I did not know about you can set multiple keys. Likely the best course of action will be to just do that from the driver.
If you have a big layer it would be really interesting to test if multi-threading makes it much faster like used here: https://github.com/geotrellis/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/s3/S3RDDWriter.scala

But I don't think the second part is critical for first implementation.